### PR TITLE
chore(samples): Code samples for Retail Tutorials. Search service samples

### PR DESCRIPTION
**Pull request by t-karasova**
This reverts commit 153c92d2

moved to interactive-tutorials dir

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-retail/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

### CODE REVIEW

**Comment by t-karasova** [comment]
@kurtisvg
Hi Kurtis,
I have created this pull request to add the code samples for Interactive Tutorials for Retail API.
You have already seen them in [#254](https://github.com/googleapis/java-retail/pull/254), which now is closed, as it was too big.

The code samples are moved to the folder "samples/interactive-tutorials" to separate them from the code samples, which are used for displaying in the static documentation pages.
It was decided, that our tutorials + code samples will be shown only by clicking the button "Guide me" in the Retail documentation page and will be opened in the Cloud Shell IDE.

In this PR I have created the code samples for Retail Search service and tests.

I have modified .kokoro configs to read the secrets from "retail-samples-secrets" secret manager.

For the tests we use our own Cloud project with preset catalog data. And I would like to ask you how should I add the service account token to our project to authorize tests?

Thank you

File: samples/interactive-tutorials/src/main/java/search/SearchWithOrdering.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  /**
+   * Call the retail search.
+   *
+   * @return SearchResponse.
+   * @throws IOException if endpoint is not provided.
+   */
+  public static SearchResponse search() throws IOException {
+    // TRY DIFFERENT FILTER EXPRESSIONS HERE:
```
Comment (chenlei1216) [inline samples/interactive-tutorials/src/main/java/search/SearchWithOrdering.java:94]: ORDER BY EXPRESSIONS?

File: samples/interactive-tutorials/src/main/java/search/SearchWithPagination.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    String pageToken = "";
+
+    SearchRequest searchRequestFirstPage = getSearchRequest("Hoodie", pageSize, offset, pageToken);
+
+    SearchResponse searchResponseFirstPage =
+        getSearchServiceClient().search(searchRequestFirstPage).getPage().getResponse();
+
+    System.out.println("Search response: " + searchResponseFirstPage);
+
+    // PASTE CALL WITH NEXT PAGE TOKEN HERE:
```
Comment (chenlei1216) [inline samples/interactive-tutorials/src/main/java/search/SearchWithPagination.java:109]: Maybe add code samples to get the next page token?

**Review by chenlei1216 — APPROVED** [review]

File: samples/interactive-tutorials/src/main/java/search/SearchWithPagination.java (hunk 2 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithPagination.java:109]: We provide a small piece of code to get the nextPageToken and use it the similar search request in the tutorial, ask user to copy and paste the code here

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * [START retail_search_for_products_with_query_parameter]
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:18]: The region tag should be in a stand alone comment, otherwise the comment below will not render correctly on the page.

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+  /**
+   * This variable describes default search placement name. Using for identify the Serving Config
+   * name.
+   */
+  private static final String DEFAULT_SEARCH_PLACEMENT_NAME =
+      DEFAULT_CATALOG_NAME + "/placements/default_search";
+
+  /** This variable describes a unique identifier to track visitors. */
+  private static final String VISITOR_ID = UUID.randomUUID().toString();
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:48]: This class should begin with a main function, which provides examples for the user to override to set any required variables. Env vars can be used from the tests to trigger the function itself. 

Please see https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#method-structure.

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+  private SearchSimpleQuery() {}
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:50]: Why is this here?

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+  /**
+   * Get search service client.
+   *
+   * @return SearchServiceClient.
+   * @throws IOException if endpoint is incorrect.
+   */
+  private static SearchServiceClient getSearchServiceClient() throws IOException {
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:58]: Why is this in a separate function? 

Please see https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#client-initialization

**Review by kurtisvg — CHANGES_REQUESTED** [review]
I reviewed a single file for now. This PR is exceptionally large (+1500) lines. 

Please see go/small-cls and break it into smaller PRs so it's easier to review.

**Comment by kurtisvg** [comment]
> I have modified .kokoro configs to read the secrets from "retail-samples-secrets" secret manager.

I don't see any modifications to kokoro in these config. Typically we don't modify the kokoro configs as they are often generated by OwlBot. @lesv or @Neenu1995 might be able to provide more info on what the preferred paths for env vars is here

> For the tests we use our own Cloud project with preset catalog data. And I would like to ask you how should I add the service account token to our project to authorize tests?

Maybe @lesv or @Neenu1995 can weigh in on this too. I think the rest of the java tests use a single project these days, but I'm not sure.

**Comment by Neenu1995** [comment]
> I have modified .kokoro configs to read the secrets from "retail-samples-secrets" secret manager.

I don't see the kokoro configs either.

> For the tests we use our own Cloud project with preset catalog data. And I would like to ask you how should I add the service account token to our project to authorize tests?

The secret key needs to be rotated periodically. It can be done using the [secret rotator](https://g3doc.corp.google.com/company/teams/cloud-devrel/dpe/guides/using-secret-rotator-service.md?cl=head). It will be easier to use the existing project to run tests unless these tests can interfere with existing test resources.

**Comment by Neenu1995** [comment]
Let us know which method you decide to use and we can work together on the infrastructure to set that up.
Also in the existing configuration, continuous tests have not been set up in kokoro. You will have to enable it if you plan to use continuous scheduled tests instead of presubmit test.

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 3 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java]: fixed

File: samples/interactive-tutorials/src/main/java/search/SearchWithOrdering.java (hunk 1 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithOrdering.java]: this file is removed to make the PR smaller - will be submitted in the following PR

File: samples/interactive-tutorials/src/main/java/search/SearchWithPagination.java (hunk 2 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithPagination.java]: this file is removed to make the PR smaller - will be submitted in the following PR

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 6 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java]: done

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 4 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java]: done

File: .kokoro/build.sh [hunk 7]
```diff
@@ -103,6 +103,30 @@ samples)
         echo "no sample pom.xml found - skipping sample tests"
     fi
     ;;
+tutorials-samples)
```
Comment (kurtisvg) [inline .kokoro/build.sh:106]: I think build.sh is a file that's generated by owlbot. If you touch it, it'll revert it back. Do we really need this change?

File: .kokoro/nightly/tutorials-samples.cfg [hunk 8]
```diff
@@ -0,0 +1,33 @@
+# Format: //devtools/kokoro/config/proto/build.proto
```
Comment (kurtisvg) [inline .kokoro/nightly/tutorials-samples.cfg:1]: Do we need a separate build for tutorials, as opposed to just samples?

File: .kokoro/build.sh (hunk 7 above)
Comment (kurtisvg) [reply .kokoro/build.sh:106]: I see you've opted out below. What's the reasoning for running this as a separate build as opposed to the rest of the samples?

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+  public static void main(String[] args) throws IOException {
+
+    String projectNumber = System.getenv("PROJECT_NUMBER");
+
+    String defaultCatalogName =
+        String.format("projects/%s/locations/global/catalogs/default_catalog", projectNumber);
+
+    String defaultSearchPlacementName = defaultCatalogName + "/placements/default_search";
+
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:42]: nit: the empty lines make this hard to read

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+import com.google.cloud.retail.v2.SearchRequest;
+import com.google.cloud.retail.v2.SearchResponse;
+import com.google.cloud.retail.v2.SearchServiceClient;
+import java.io.IOException;
+import java.util.UUID;
+
+public class SearchSimpleQuery {
+
+  public static void main(String[] args) throws IOException {
+
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:35]:
```suggestion
    // TODO(developer): Replace these variables before running the sample.
```

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+    search(defaultSearchPlacementName);
+  }
+
+  /**
+   * Call the retail search.
+   *
+   * @return SearchResponse.
+   * @throws IOException if endpoint is not provided.
+   */
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:51]: Please [avoid using java doc for method comments](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#method-comment).

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+            .setQuery(query)
+            .setVisitorId(visitorId)
+            .setPageSize(pageSize)
+            .build();
+
+    System.out.println("Search request: " + searchRequest);
+
+    return searchRequest;
+  }
+}
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java:89]: I'm not really sure why these are two different methods - are they different snippets or should they just be combined? [They also should avoid having a return](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#method-structure).

File: samples/pom.xml [hunk 13]
```diff
@@ -16,9 +16,9 @@
     Removing or replacing it should not affect the execution of the samples in anyway.
   -->
   <parent>
-    <groupId>com.google.cloud.samples</groupId>
-    <artifactId>shared-configuration</artifactId>
-    <version>1.2.0</version>
+    <artifactId>google-cloud-retail-parent</artifactId>
```
Comment (kurtisvg) [inline samples/pom.xml:19]: Why are you overriding shared-configuration? This should be the parent for all samples

**Review by kurtisvg — CHANGES_REQUESTED** [review]

**Comment by lesv** [comment]
# ci/ Units (8)
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running search.SearchWithFacetSpecTest
Search request: placement: "projects/null/locations/global/catalogs/default_catalog/placements/default_search"
query: "Tee"
visitor_id: "284d23e5-6f96-4efd-b57b-95d4e061fba4"
page_size: 10
facet_specs {
  facet_key {
    key: "colorFamilies"
  }
}

Error:  Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 12.411 s <<< FAILURE! - in search.SearchWithFacetSpecTest
Error:  search.SearchWithFacetSpecTest.testSearchWithFacetSpec  Time elapsed: 7.36 s  <<< ERROR!
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at search.SearchWithFacetSpecTest.testSearchWithFacetSpec(SearchWithFacetSpecTest.java:71)

Error:  search.SearchWithFacetSpecTest.testOutput  Time elapsed: 5.027 s  <<< FAILURE!
java.lang.AssertionError
	at search.SearchWithFacetSpecTest.testOutput(SearchWithFacetSpecTest.java:61)

[INFO] Running search.SearchSimpleQueryTest
Search request: placement: "projects/null/locations/global/catalogs/default_catalog/placements/default_search"
query: "Hoodie"
visitor_id: "f9257b3e-e41a-40fe-9583-82e808cf33d9"
page_size: 10

Error:  Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 10.056 s <<< FAILURE! - in search.SearchSimpleQueryTest
Error:  search.SearchSimpleQueryTest.testSearchSimpleQuery  Time elapsed: 4.967 s  <<< ERROR!
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at search.SearchSimpleQueryTest.testSearchSimpleQuery(SearchSimpleQueryTest.java:69)

Error:  search.SearchSimpleQueryTest.testOutput  Time elapsed: 5.085 s  <<< FAILURE!
java.lang.AssertionError
	at search.SearchSimpleQueryTest.testOutput(SearchSimpleQueryTest.java:61)

[INFO] Running search.SearchWithBoostSpecTest
Search request: placement: "projects/null/locations/global/catalogs/default_catalog/placements/default_search"
query: "Tee"
visitor_id: "3ed482de-f9b7-40c5-9da5-0d4a298bc0cc"
page_size: 10
boost_spec {
  condition_boost_specs {
    condition: "(colorFamilies: ANY(\"Blue\"))"
  }
}

Error:  Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 10.236 s <<< FAILURE! - in search.SearchWithBoostSpecTest
Error:  search.SearchWithBoostSpecTest.testSearchWithBoostSpec  Time elapsed: 5.055 s  <<< ERROR!
java.io.IOException: The Application Default Credentials are not available. They are available if running in Google Compute Engine. Otherwise, the environment variable GOOGLE_APPLICATION_CREDENTIALS must be defined pointing to a file defining the credentials. See https://developers.google.com/accounts/docs/application-default-credentials for more information.
	at search.SearchWithBoostSpecTest.testSearchWithBoostSpec(SearchWithBoostSpecTest.java:69)

Error:  search.SearchWithBoostSpecTest.testOutput  Time elapsed: 5.176 s  <<< FAILURE!
java.lang.AssertionError
	at search.SearchWithBoostSpecTest.testOutput(SearchWithBoostSpecTest.java:61)

[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    SearchSimpleQueryTest.testOutput:61
Error:    SearchWithBoostSpecTest.testOutput:61
Error:    SearchWithFacetSpecTest.testOutput:61
Error:  Errors: 
Error:    SearchSimpleQueryTest.testSearchSimpleQuery:69 » IO The Application Default Cr...
Error:    SearchWithBoostSpecTest.testSearchWithBoostSpec:69 » IO The Application Defaul...
Error:    SearchWithFacetSpecTest.testSearchWithFacetSpec:71 » IO The Application Defaul...
[INFO] 
Error:  Tests run: 6, Failures: 3, Errors: 3, Skipped: 0
```
# ci / dependencies

```
[INFO] --- maven-dependency-plugin:3.1.2:analyze (default-cli) @ retail-install-without-bom ---
Warning:  Unused declared dependencies found:
Warning:     com.google.cloud:google-cloud-retail:jar:2.0.7:compile
Warning:     junit:junit:jar:4.13.2:test
Warning:     com.google.truth:truth:jar:1.1.3:test
```

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Google Cloud Retail Parent 2.0.9-SNAPSHOT .......... SUCCESS [  3.514 s]
[INFO] proto-google-cloud-retail-v2 2.0.9-SNAPSHOT ........ SUCCESS [  0.830 s]
[INFO] grpc-google-cloud-retail-v2 2.0.9-SNAPSHOT ......... SUCCESS [  0.230 s]
[INFO] Google Cloud Retail 2.0.9-SNAPSHOT ................. SUCCESS [  0.492 s]
[INFO] Google Cloud Retail BOM 2.0.9-SNAPSHOT ............. SUCCESS [  0.015 s]
[INFO] Google Cloud Retail Samples Parent 0.0.1-SNAPSHOT .. SUCCESS [  0.087 s]
[INFO] Google Cloud Retail Install Without Bom 0.0.1-SNAPSHOT FAILURE [  0.225 s]
[INFO] Google Cloud Retail Snapshot Samples 0.0.1-SNAPSHOT  SKIPPED
[INFO] Google Cloud Retail Snippets 0.0.1-SNAPSHOT ........ SKIPPED
[INFO] Google Cloud Retail Interactive Tutorials 0.0.1-SNAPSHOT SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```

File: samples/pom.xml (hunk 13 above)
Comment (sborisenkox) [reply samples/pom.xml]: I have tried to set up configuration because IDE (IntelliJ IDEA) is not recognized Maven modules properly. I can't run my code if Maven modules relations are not configured.

File: .kokoro/nightly/tutorials-samples.cfg (hunk 8 above)
Comment (t-karasova) [reply .kokoro/nightly/tutorials-samples.cfg:1]: We with Neenu have discussed the approach, she suggested the tests to be run with the Nightly Job. So I have added this config and new JOB_TYPE to run just tutorial-samples.

Is that fine or you would like to implement some other approach?

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 10 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java]: fixed

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 9 above)
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java]: fixed

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 11 above)
Comment (sborisenkox) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java]: Changed.

File: samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java (hunk 12 above)
Comment (sborisenkox) [reply samples/interactive-tutorials/src/main/java/search/SearchSimpleQuery.java]: Combined two methods in one.

File: samples/pom.xml (hunk 13 above)
Comment (t-karasova) [reply samples/pom.xml]: reverted

File: .kokoro/nightly/tutorials-samples.cfg (hunk 8 above)
Comment (kurtisvg) [reply .kokoro/nightly/tutorials-samples.cfg:1]: There is a ["nightly" samples](https://fusion.corp.google.com/projectanalysis/summary/KOKORO/prod:cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-retail%2Fnightly%2Fsamples?search_pattern=cloud-devrel%2Fclient-libraries%2Fjava%2Fjava-retail%2F&project_types=KOKORO&include_inactive_projects=false) build already happening. Do we need a separate job for the tutorials, or can we just keep them all in one job?

File: samples/install-without-bom/pom.xml [hunk 14]
```diff
@@ -11,7 +11,7 @@
     The parent pom defines common style checks and testing strategies for our samples.
     Removing or replacing it should not affect the execution of the samples in anyway.
   -->
-  <parent>
```
Comment (kurtisvg) [inline samples/install-without-bom/pom.xml:14]: nit: this should stay 2 spaces

File: samples/interactive-tutorials/pom.xml [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+  <artifactId>retail-interactive-tutorials</artifactId>
+  <packaging>jar</packaging>
+  <name>Google Cloud Retail Interactive Tutorials</name>
+  <url>https://github.com/googleapis/java-retail</url>
+
+ <parent>
+    <groupId>com.google.cloud.samples</groupId>
+    <artifactId>shared-configuration</artifactId>
+    <version>1.2.0</version>
+  </parent>
```
Comment (kurtisvg) [inline samples/interactive-tutorials/pom.xml:16]: Please include the [comment here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md#project-dependencies).

File: samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+    String projectNumber = System.getenv("PROJECT_NUMBER");
+    String defaultCatalogName =
+        String.format("projects/%s/locations/global/catalogs/default_catalog", projectNumber);
+    String defaultSearchPlacementName = defaultCatalogName + "/placements/default_search";
+
+    search(defaultSearchPlacementName);
+  }
+
+  public static SearchResponse search(String defaultSearchPlacementName) throws IOException {
+    // TRY DIFFERENT CONDITIONS HERE:
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java:47]: nit: it's preferable to link to any documentation on what these values are and what they can do

File: samples/interactive-tutorials/src/test/java/search/SearchSimpleQueryTest.java [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+    String projectNumber = System.getenv("PROJECT_NUMBER");
+    String defaultCatalogName =
+        String.format("projects/%s/locations/global/catalogs/default_catalog", projectNumber);
+    defaultSearchPlacementName = defaultCatalogName + "/placements/default_search";
+    Process exec =
+        Runtime.getRuntime()
+            .exec("mvn compile exec:java -Dexec.mainClass=search.SearchSimpleQuery");
+    StreamGobbler streamGobbler = new StreamGobbler(exec.getInputStream());
+    Future<String> stringFuture = Executors.newSingleThreadExecutor().submit(streamGobbler);
+    output = stringFuture.get();
```
Comment (kurtisvg) [inline samples/interactive-tutorials/src/test/java/search/SearchSimpleQueryTest.java:44]: What's going on here?

Comment (kurtisvg) [reply samples/interactive-tutorials/src/test/java/search/SearchSimpleQueryTest.java:44]: If you're just trying to get the output, I suggest you try this method: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/kms/src/test/java/kms/SnippetsIT.java#L108-L116

File: samples/pom.xml [hunk 18]
```diff
@@ -43,6 +45,13 @@
           <skip>true</skip>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-checkstyle-plugin</artifactId>
+        <configuration>
+          <skip>true</skip>
+        </configuration>
+      </plugin>
```
Comment (kurtisvg) [inline samples/pom.xml:54]: Please don't do this. Checkstyle is there for a reason.

**Review by kurtisvg — CHANGES_REQUESTED** [review]

**Review by kurtisvg — CHANGES_REQUESTED** [review]

File: samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java [hunk 19]
```diff
@@ -0,0 +1,69 @@
...
+            .addFacetSpecs(facetSpec)
+            .setPageSize(pageSize)
+            .build();
+    System.out.println("Search request: " + searchRequest);
+
+    SearchResponse searchResponse =
+        SearchServiceClient.create().search(searchRequest).getPage().getResponse();
+    System.out.println("Search response: " + searchResponse);
+
+    return searchResponse;
```
[the remark this answers is not in the document]
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java:65]: We need this return values for tests. In the test we call this method and check the response

File: samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+        SearchRequest.newBuilder()
+            .setPlacement(defaultSearchPlacementName)
+            .setQuery(searchQuery)
+            .setVisitorId(visitorId)
+            .addFacetSpecs(facetSpec)
+            .setPageSize(pageSize)
+            .build();
+    System.out.println("Search request: " + searchRequest);
+
+    SearchResponse searchResponse =
```
[the remark this answers is not in the document]
Comment (t-karasova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java:61]: Do you mean we need to put our code (client initialization and client call) into try-catch, like this:

private static SearchResponse getSearchResponse(String query) throws IOException {
    try (SearchServiceClient client = SearchServiceClient.create()) {
      SearchRequest request = SearchRequest.newBuilder().setQuery(query).build();
      SearchResponse response = client.search(request);
      return response;
    }
  }

File: .kokoro/nightly/tutorials-samples.cfg (hunk 8 above)
Comment (Neenu1995) [reply .kokoro/nightly/tutorials-samples.cfg:1]: The reasoning was to keep interactive tutorials separate from the regular samples for ease of code maintenance. Unless there is a strong recommendation against it, let's keep it separate.

File: .kokoro/build.sh (hunk 7 above)
Comment (t-karasova) [reply .kokoro/build.sh:106]: Please see the Neenu comment below

File: samples/install-without-bom/pom.xml (hunk 14 above)
Comment (sborisenkox) [reply samples/install-without-bom/pom.xml:14]: Fixed.

File: samples/interactive-tutorials/pom.xml (hunk 15 above)
Comment (sborisenkox) [reply samples/interactive-tutorials/pom.xml:16]: Fixed.

File: samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java (hunk 20 above)
[the remark this answers is not in the document]
Comment (sborisenkox) [reply samples/interactive-tutorials/src/main/java/search/SearchWithFacetSpec.java:61]: Changed.

File: samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java (hunk 16 above)
Comment (tkarasyova) [reply samples/interactive-tutorials/src/main/java/search/SearchWithBoostSpec.java:47]: We have the tutorials, published in google3, in that tutorials we lead user through the service features description and provide different values which can demonstrate the feature in a particular catalog data

**Comment by t-karasova** [comment]
CLosed due to the CLA issue - the code changes are duplicated to the  PR #289
